src/misc/HtmlSanitizer.ts

```diff
@@ -314,15 +317,27 @@ export class HtmlSanitizer {
 				} else if (config.blockExternalContent && attribute.name === "srcset") {
 					this.externalContent++
 
+					htmlNode.setAttribute("draft-srcset", attribute.value)
 					htmlNode.removeAttribute("srcset")
 					htmlNode.setAttribute("src", PREVENT_EXTERNAL_IMAGE_LOADING_ICON)
 					htmlNode.style.maxWidth = "100px"
-				} else if (config.blockExternalContent && !attribute.value.startsWith("data:") && !attribute.value.startsWith("cid:")) {
+				} else if (
+					config.blockExternalContent &&
+					!attribute.value.startsWith("data:") &&
+					!attribute.value.startsWith("cid:") &&
+					!attribute.name.startsWith("draft-")
+				) {
 					this.externalContent++
 
+					htmlNode.setAttribute("draft-" + attribute.name, attribute.value)
 					attribute.value = PREVENT_EXTERNAL_IMAGE_LOADING_ICON
 					htmlNode.attributes.setNamedItem(attribute)
 					htmlNode.style.maxWidth = "100px"
+				} else if (!config.blockExternalContent && attribute.name.startsWith("draft-")) {
+					const originalAttribute = attribute.name.split("-")
```

Comment:
this seems like a wrong place to do it. We don't do this with inline images, we have another pass afterwards

Reply:
nvm, it needs to know about these attributes anyway but I don't think we should put *anything* that starts with `draft-` and blindly assign it, can we put both cases here explicitly?

Reply:
Sure, do we need to care about picture html tag? Or just img with src/srcset?

Reply:
Picture just includes images in it, doesn't it?

Reply:
Yes, it's already covered, we don't need a specially treatment just for it